modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts

```diff
@@ -0,0 +1,292 @@
+/**
+ * Zero Knowledge Range Proofs as described in (Two-party generation of DSA signatures)[1].
+ * [1]: https://reitermk.github.io/papers/2004/IJIS.pdf
+ */
+import { createHash } from 'crypto';
+import BaseCurve from '../../curves';
+import { PublicKey } from 'paillier-bigint';
+import { bitLength, primeSync, randBitsSync, randBetween } from 'bigint-crypto-utils';
+import { gcd, modPow } from 'bigint-mod-arith';
+import { NTilde, RangeProof, RangeProofWithCheck } from './types';
+import { bigIntFromBufferBE, bigIntToBufferBE } from '../../util';
+
+function generateModulus(bitlength: number): bigint {
+  let n, p, q;
+  do {
+    p = primeSync(Math.floor(bitlength / 2) + 1);
```

Comment:
why `primeSync` and not `prime`?  the lib recommends using `prime` since it uses workers and parallelizes it.

Reply:
This would force signShare and signConvert to become async methods and introduce many more breaking changes.

Reply:
Given how slow our tests are on this branch, I wonder if this is worth looking into? I'm going to try and make those things async and see if it's feasible